### ISSUE TITLE
[Jobs] Timestamp field switches between seconds and milliseconds

### ISSUE THREAD

**Issue by ijrsvt** [ray-project/ray#16452]
### What is the problem?

When I create a Job (and it is still running) the `timestamp` field is in seconds, but when it is terminated, the `timestamp` field is in milliseconds.

*Ray version and other system information (Python version, TensorFlow version, OS):*

### Reproduction (REQUIRED)
Please provide a short code snippet (less than 50 lines if possible) that can be copy-pasted to reproduce the issue. The snippet should have **no external library dependencies** (i.e., use fake or mock data / environments):

Open a terminal and do the following:
```
import ray
ray.init()
ray.state.jobs()
"""
{'JobID': '04000000',
  'DriverIPAddress': '172.31.50.0',
  'DriverPid': 1222,
  'Timestamp': 1623797674,
  'StartTime': 1623797674637,
  'EndTime': 0,
  'IsDead': False}
"""
```
Close the terminal and do:
```
import ray
ray.init()
ray.state.jobs()
"""
 {'JobID': '04000000',
  'DriverIPAddress': '',
  'DriverPid': -1,
  'Timestamp': 1623797768754,
  'StartTime': 1623797674637,
  'EndTime': 1623797768754,
  'IsDead': True}
"""

```

- [ ] I have verified my script runs in a clean environment and reproduces the issue.
- [ ] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/master/installation.html).

**Comment by ijrsvt** [comment]
Specifically we use seconds [here](https://github.com/ray-project/ray/blob/412085dea779d44b8980e79fb1fe719fb1eb9bed/src/ray/raylet/node_manager.cc#L1112) on JobCreation and in [milliseconds on completion](https://github.com/ray-project/ray/blob/412085dea779d44b8980e79fb1fe719fb1eb9bed/src/ray/gcs/gcs_server/gcs_job_manager.cc#L72).

**Comment by wuisawesome** [comment]
yikes :/ we should prob just standardize on milliseconds?

**Comment by ijrsvt** [comment]
@wuisawesome I have no preference for seconds vs milliseconds, but standardization would be great

**Comment by wuisawesome** [comment]
It would be nice to be more granular than seconds, but standardizing between milliseconds and nanoseconds everywhere will be a PITA.

**Comment by AmeerHajAli** [comment]
@wuisawesome , can you please make a fix for this and maybe make it part of 1.4.1 release?
@ijrsvt maybe we can patch it.

**Comment by ijrsvt** [comment]
> It would be nice to be more granular than seconds, but standardizing between milliseconds and nanoseconds everywhere will be a PITA.

I agree that standardizing everywhere is hard, but this is specifically about one field being set as both. This should be relatively easy to fix, right?

**Comment by wuisawesome** [comment]
@ both of you, yes.
